unsafe/src/main/java/org/apache/spark/unsafe/bitset/BitSetMethods.java

```diff
@@ -71,8 +72,10 @@ public static boolean isSet(Object baseObject, long baseOffset, int index) {
    * Returns {@code true} if any bit is set.
    */
   public static boolean anySet(Object baseObject, long baseOffset, long bitSetWidthInBytes) {
-    for (int i = 0; i <= bitSetWidthInBytes; i++) {
-      if (PlatformDependent.UNSAFE.getByte(baseObject, baseOffset + i) != 0) {
+    assert bitSetWidthInBytes % SIZE_OF_LONG == 0;
+    int widthInLong = (int)(bitSetWidthInBytes / SIZE_OF_LONG);
+    for (int i = 0; i <= widthInLong; i++) {
+      if (PlatformDependent.UNSAFE.getLong(baseObject, baseOffset + i) != 0) {
```

Comment:
I don't think that this is correct, since `i` loops over words but the second argument to `getLong` is measured in bytes.  You could do `i * 8` instead.

Reynold pointed out that there might be faster micro-optimizations tricks that can be played here; see https://groups.google.com/d/msg/mechanical-sympathy/k0qd7dLHFQE/gsTFSjP6MVMJ